### PR TITLE
Feat/zs3server

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
- Installs zs3server in CICD pipeline

### CODE REVIEW

**Review by shahnawaz-creator — APPROVED** [review]

File: deploy-0chain/action.yml [hunk 1]
```diff
@@ -447,6 +447,28 @@ runs:
           echo "==========================================================="
         fi
 
+    - name: "Install Zs3Server"
```
Comment (stewartie4) [inline deploy-0chain/action.yml:450]: minor: rename to match what this step does

File: deploy-0chain/action.yml [hunk 2]
```diff
@@ -447,6 +447,28 @@ runs:
...
+      shell: 'script --return --quiet --command "bash {0}"'
+      run: |
+        ./zwallet faucet --methodName "pour" --tokens 9 --input "{}" --silent
+        ./zbox newallocation --lock 5 --data 2 --parity 2 --silent | grep -o "[a-z0-9]\{64\}" > allocation.txt
+        echo "Allocation hash: $(cat allocation.txt)"
```
Comment (stewartie4) [inline deploy-0chain/action.yml:455]: remove?

File: deploy-0chain/action.yml [hunk 3]
```diff
@@ -447,6 +447,28 @@ runs:
...
+      shell: 'script --return --quiet --command "bash {0}"'
+      run: |
+        helm upgrade --install --wait \
+          --set zs3server.hostName=dev-${{ env.RUNNER_NUMBER }} \
+          --set zs3server.host=devnet-0chain.net \
+          --set zs3server.client.image.tag=v0.0.2 \
+          --set zs3server.logsearchapi.image.tag=v0.0.1 \
+          --set zs3server.minio.image.tag=v0.0.1 \
+          --set zs3server.minio.config.zcn_config.block_worker=https://${{ env.NETWORK_URL }}/dns \
+          --set zs3server.minio.config.allocation_1=${{ env.ALLOCATION }} \
```
Comment (stewartie4) [inline deploy-0chain/action.yml:469]: what is this for out of interest?

**Review by stewartie4 — APPROVED** [review]
Happy to approve, but some minor comments if you're in here again

File: deploy-0chain/action.yml (hunk 1 above)
Comment (Kishan-Dhakan) [reply deploy-0chain/action.yml:450]: Done

File: deploy-0chain/action.yml (hunk 2 above)
Comment (Kishan-Dhakan) [reply deploy-0chain/action.yml:455]: It's necessary, because allocation is needed in next step

File: deploy-0chain/action.yml (hunk 3 above)
Comment (Kishan-Dhakan) [reply deploy-0chain/action.yml:469]: It's a migration tool allowing clients to migrate data from S3 to an allocation

**Comment by Kishan-Dhakan** [comment]
Thanks, resolved @stewartie4

**Pull request merged**